Update article: View, edit, and delete cookies (#2164)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/storage/cookies.md

ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge

> ms.date: 06/08/2022

Reviewer: ```suggestion
ms.date: 09/08/2022
```

Author: @mikehoffms you probably noticed, there are two new fields in the cookies table that are blank because I don't know what/why they are.

Reviewer: @dan-wesley Pls close this Outdated conversation if resolved.

Reviewer: @dan-wesley (or @captainbrosset )  Please close this "Outdated" conversation as Resolved if this has been resolved in the latest files.

<!-- ====================================================================== -->
## Open the Cookies pane

> 1. Open DevTools on the webpage you want to edit. To open DevTools, use one of the techniques described in [Open DevTools](/microsoft-edge/devtools-guide-chromium/open).

Reviewer: ```suggestion
1. Open DevTools on the webpage you want to edit.  For example, right-click the page and then select **Inspect**, or press F12.  For additional ways, see [Open DevTools](/microsoft-edge/devtools-guide-chromium/open).
```
Provide at least one way, inline, in order to avoid having to leave the page.

Reviewer: Probably delete "For additional ways, see [Open DevTools](/microsoft-edge/devtools-guide-chromium/open)."  Not needed, not the focus.

> 2. Click the **Application** tab to open the **Application** panel. The **Manifest** pane opens:

Reviewer 2: ```suggestion
2. Click the **Application** tab to open the **Application** tool. The **Manifest** pane opens:
```

>    ![The Manifest pane.](cookies-images/pick-application-no-manifest.png)

Reviewer: ```suggestion
   ![The Manifest pane](cookies-images/pick-application-no-manifest.png)
```

> 2. Click the **Application** icon (![Application icon in Activity Bar](images/application-storage-cookies-application-icon.png)) in the Activity Bar or select "Application" in the Quick View dropdown list to open the **Application** panel.  The **Manifest** pane opens:

Reviewer: "icon in Activity Bar" - pls check to see if wording & captures should show Focus Mode UI or not

Reviewer: @dan-wesley pls close this thread is done/feasible.

3. Under **Storage**, expand **Cookies**, then select an origin:

>    ![The Cookies pane.](cookies-images/open-cookies-select-source.png)

Reviewer: ```suggestion
   ![The Cookies pane](cookies-images/open-cookies-select-source.png)
```

>    ![The Cookies pane.](images/application-storage-cookies-expand-cookies-select-origin.png)

Reviewer: ```suggestion
   ![The Cookies pane.](cookies-images/application-storage-cookies-expand-cookies-select-origin.png)
```
moving to dedicated image dirs

Reviewer 2: Nice! Although I can see one problem with doing this now: it will make it harder to merge the incoming changes from the `focus-mode-docs-bash` branch. We have tons of image changes coming in that branch, and we're not quite ready to merge them yet. So, we have to maintain the branch in parallel to main for a while still. Therefore, any changes to the path of images in main risk introducing merge conflicts in the focus-mode-docs-bash.
I would prefer if we didn't change any image paths until we merge the focus mode branch. And once that's done, we can move all images automatically in one go, with a script.

Reviewer: (This is an "Outdated" comment thread.)  @captainbrosset - it's your call, I suggest keeping the original filenames, for tracking.  @dan-wesley

Reviewer: Dan (or Capt.) pls close this "Outdated" thread if feasible.

Reviewer 2: Yes, I would prefer it if we could keep the same image names and paths if possible. This way when merging the `focus-mode-docs-bash` branch, it will just be a question of choosing which image is best.

Reviewer: @dan-wesley (or @captainbrosset )  Please close this "Outdated" conversation as Resolved if this has been resolved in the latest files.

*  **Size**.  The size, in bytes, of the cookie.

> *  **HttpOnly**.  If true, this field indicates that the cookie should only be used over HTTP, and JavaScript modification isn't allowed.  See [HttpOnly cookies](https://developer.mozilla.org/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies).

Reviewer: ```suggestion
*  **HttpOnly**.  If `true`, this field indicates that the cookie should only be used over HTTP, and JavaScript modification isn't allowed.  See [HttpOnly cookies](https://developer.mozilla.org/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies).
```
if ui, use bold instead of backticks.  User-entered text in a ui is bolded. Editing code in a code file is backticks.

> *  **Secure**.  If true, this field indicates that the cookie must be sent to the server only over a secure, HTTPS connection.  See [Secure cookies](https://developer.mozilla.org/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies).

Reviewer: ```suggestion
*  **Secure**.  If `true`, this field indicates that the cookie must be sent to the server only over a secure, HTTPS connection.  See [Secure cookies](https://developer.mozilla.org/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies).
```
backtick if editing code, bold if using UI

*  **SameSite**.  Contains `strict` or `lax` if the cookie is using the experimental [Samesite](https://developer.mozilla.org/docs/Web/HTTP/Cookies#SameSite_cookies) attribute.

> *  **SameParty**. This attribute provides web developers a means to annotate cookies that are allowed to be set or sent in same-party, cross-site contexts. 

Reviewer: ```suggestion
*  **SameParty**. This attribute provides web developers a means to annotate cookies that are allowed to be set or sent in "same-party, cross-site" contexts. 
```

> *  **Partition Key**. Partition key is the property or path within documents that can be used to distribute data. Data having the same partition key value are logically grouped together and stored in the same physical partition.

Reviewer: ```suggestion
*  **Partition Key**. A _partition key_ is the property or path within a document that can be used to distribute data within a partition. Multiple pieces of data that are spread throughout a document and that have the same partition key value are logically grouped together and stored in the same physical partition.
```
pls tech-review "within a partition".  Clarify "to distribute data"; what's the specific context/objective?  @captainbrosset

Reviewer 2: I'm not an expert on this at all. So far, the simplest explanation I've read about this is from this section: https://developer.chrome.com/blog/chips-origin-trial/#what-is-chips

I think the partition key field in DevTools is meant to contain URLs (or at least domains, although I've not been able to test it successfully).
Partitioned cookies allow third-party websites to set cookies on client machines, but in a way that prevents other domains to read them.
The "Partition key" column in DevTools is way for users to see when a cookie is partitioned, and if so, which domain it belongs to.

Based on my understanding I think this part of the docs needs to be changed. But, again, I'm new to this, so can't help much more.

Author: @mikehoffms, @captainbrosset 
This is an area that I know nothing about. My initial definition is a copy from https://docs.microsoft.com/en-us/azure/cosmos-db/partitioning-overview.

*  **Priority**.  Contains `low`, `medium` (default), or `high` if the cookie is using the deprecated [Cookie Priority](https://bugs.chromium.org/p/chromium/issues/detail?id=232693) attribute.

<!-- ====================================================================== -->
## Filter cookies

To filter cookies by **Name** or **Value**, use the **Filter** text box:

> ![Filtering out any cookies that don't contain the text ID.](cookies-images/filter-cookies-name.png)

Reviewer: ```suggestion
![Filtering out any cookies that don't contain the text ID](cookies-images/filter-cookies-name.png)
```

**Note:** Filtering by other fields isn't supported.

<!-- ====================================================================== -->
## Edit a cookie

The **Name**, **Value**, **Domain**, **Path**, and **Expires / Max-Age** fields are editable.  Double-click a field to edit it:

![Setting the name of a cookie to 'DEVTOOLS!'](cookies-images/rename-cookie.png)

<!-- ====================================================================== -->
## Delete cookies

To delete a specific cookie, click a cookie and then click **Delete Selected** (![Delete Selected](cookies-images/delete-cookie-icon.png)):

![Deleting a specific cookie.](cookies-images/delete-selected-cookie.png)

> To delete all cookies, click **Clear All** (![Clear All.](cookies-images/clear-all-cookies-icon.png)):

Reviewer: ```suggestion
To delete all cookies, click **Clear All** (![Clear All](cookies-images/clear-all-cookies-icon.png)):
```

> ![Clearing all cookies.](cookies-images/clear-all-cookies.png)

Reviewer: ```suggestion
![Clearing all cookies](cookies-images/clear-all-cookies.png)
```

*  **HttpOnly**.  If true, this field indicates that the cookie should only be used over HTTP, and JavaScript modification isn't allowed.  See [HttpOnly cookies](https://developer.mozilla.org/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies).

*  **Secure**.  If true, this field indicates that the cookie must be sent to the server only over a secure, HTTPS connection.  See [Secure cookies](https://developer.mozilla.org/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies).

*  **SameSite**.  Contains `strict` or `lax` if the cookie is using the experimental [Samesite](https://developer.mozilla.org/docs/Web/HTTP/Cookies#SameSite_cookies) attribute.

*  **SameParty**. 

*  **Partition Key**.

*  **Priority**.  Contains `low`, `medium` (default), or `high` if the cookie is using the deprecated [Cookie Priority](https://bugs.chromium.org/p/chromium/issues/detail?id=232693) attribute.

<!-- ====================================================================== -->
## Filter cookies

To filter cookies by **Name** or **Value**, use the **Filter** text box:

![Filtering out any cookies that don't contain the text ID.](images/application-storage-cookies-filter-cookies.png)

> > [!NOTE]

Reviewer: https://docs.microsoft.com/en-us/contribute/markdown-reference#alerts-note-tip-important-caution-warning
"Avoid notes, tips, and important boxes. Readers tend to skip over them. It's better to put that info directly into the article text."

Several alternatives:
h4 with a label/heading describing the point.
**Note:** blah

Author: @mikehoffms Got it, thanks

Reviewer: @dan-wesley Pls close this "Outdated" conversation - been resolved.

Reviewer: @dan-wesley (or @captainbrosset )  Please close this "Outdated" conversation as Resolved if this has been resolved in the latest files.
